Adding output functionality

lpessoa opened the pull request:
Fixes #6109
Output flag allows one to download to a specific file location or even redirect to output using '-' as argument.

<!--
  Thank you for contributing to GitHub CLI!
  To reference an open issue, please write this in your description: `Fixes #NUMBER`
-->

lpessoa committed:
Adding output functionality

Output flag allows one to download to a specific file location or event redirect to output using '-' as argument.

mislav reviewed (commented):
Looks like a good start; thank you! Let us know when this is ready for final review 👍

lpessoa committed:
adding tests and using iostreams out as stdout

Using iostreams value from command options when redirecting to stdout as required.
Included tests for 'output' option.

lpessoa commented:
@mislav did some cleanup on the code and added tests.

mislav committed:
Merge remote-tracking branch 'origin' into lp-release-output

mislav committed:
simplify writing release assets to destination

mislav committed:
OutputFile

mislav reviewed (approved):
Thank you for the feature!

I was concerned about the number of different arguments needing to be passed down to `downloadAsset()` after your change, so I've pushed a tweak that introduces a `destinationWriter` to abstract away the concept of writing a downloaded asset into a file or to stdout.

mislav committed:
fix windows

lpessoa commented:
@mislav Thanks for the changes. I tend to make changes as small as possible but in this case the options wrapper was for sure a cleaner approach. 😁
